release_util/management/commands/generate_history.py

```diff
@@ -0,0 +1,133 @@
+"""
+Management command to populate initial history.
+"""
+import logging
+import time
+from datetime import datetime
+from django.apps import apps
+from django.core.management.base import BaseCommand
+from django.db import connection, transaction
+
+get_model = apps.get_model
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Populate initial history for models using django-simple-history.
+    Example usage:
+    $ ./manage.py lms generate_history --models organizations.organization entitlements.courseentitlement --batchsize 1000 --sleep_between 1 --settings=devstack
+    """
+
+    help = (
+        "Populates the corresponding historical records with"
+        "the current state of records using django-simple-history"
+        "which do not have a historical record yet."
+    )
+
+    DEFAULT_BATCH_SIZE = 200
+    DEFAULT_SLEEP_BETWEEN_INSERTS = 1
+    HISTORY_DATE = datetime.today().strftime('%Y-%m-%d')
+    HISTORY_USER_ID = 'NULL'
+    HISTORY_CHANGE_REASON = 'initial history population'
+
+    def add_arguments(self, parser):
+        super(Command, self).add_arguments(parser)
+
+        parser.add_argument("--tables", nargs="*", type=str)
+
+        parser.add_argument(
+            '--sleep_between',
+            default=self.DEFAULT_SLEEP_BETWEEN_INSERTS,
+            type=float,
+            help='Seconds to sleep between chunked inserts.'
+        )
+
+        parser.add_argument(
+            "--batchsize",
+            action="store",
+            default=self.DEFAULT_BATCH_SIZE,
+            type=int,
+            help="Maximum number of history rows to insert in each batch.",
+        )
+
+
+    def columns_from_schema(self, cursor, table):
+        query = u"""
+            SELECT
+                column_name
+            FROM information_schema.columns
+            WHERE table_name='{}'
+            ORDER BY ordinal_position
+            """.format(table)
+        cursor.execute(query)
+        columns = [column[0] for column in cursor.fetchall()]
+        return columns
+
+    def handle(self, *args, **options):
+        tables = options.get("tables", [])
+        increment = options['batchsize']
+        sleep_between = options['sleep_between']
+
+        for table in tables:
+            historical_table = '_historical'.join(table.rsplit('_', 1))
+            with connection.cursor() as cursor:
+                query = u"""
+                    SELECT
+                        MIN(t.id),
+                        MAX(t.id)
+                    FROM {table} t
+                    LEFT JOIN {historical_table}
+                        ON t.id = {historical_table}.id
+                    WHERE {historical_table}.id IS NULL
+                    """.format(
+                        table=table,
+                        historical_table=historical_table,
+                )
+                cursor.execute(query)
```

Comment:
I think you mentioned this elsewhere, but it might be good to include a comment about the assumption of the name of the historical table.  If it doesn't exist, I assume this like will just throw an exception and exit if the table isn't found.   Comment should probably go on line 74?